src/components/Home.js

```diff
@@ -0,0 +1,161 @@
+import React from "react";
+import {connect} from 'react-redux';
+import {withRouter} from 'react-router';
+import {getOffers,getCategories} from '../actions/homeActions';
+import {resetCartData} from '../actions/productActions';
+import SvgLoading from '../presentations/SvgLoading';
+import Header from '../presentations/Header';
+import Footer from '../presentations/Footer';
+import {getViewPortDimensions} from '../utils';
+import LazyLoad, { forceCheck } from 'react-lazyload';
+import Slider from 'react-slick';
+import { Modal } from "react-responsive-modal";
+import Cart from '../presentations/Cart';
+import Constants from '../constants';
+import toastr from 'toastr';
+class Home extends React.Component {
+    constructor(props){
+        super(props);
+        this.state={
+          currentInstance: this,
```

Comment:
saving instance of the class in the state just so you can call a non-static method from inside a static method. This is anti pattern